app/components/works/show/manage_sharing_button_component.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module Works
```

Comment:
Encapsulating this button in a component made it a little cleaner and easier to test.